src/Npgsql/NpgsqlActivitySource.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Npgsql.Internal;
+using System;
+using System.Diagnostics;
+using System.Net;
+
+namespace Npgsql
+{
+    static class NpgsqlActivitySource
+    {
+#if NET5_0_OR_GREATER
+        static readonly ActivitySource Source = new("Npgsql");
+
+        public static bool IsEnabled => Source.HasListeners();
+
+        public static IDisposable? CommandStart(NpgsqlConnector connector, string sql)
+        {
+            var settings = connector.Settings;
+            var activity = Source.StartActivity(settings.Database!, ActivityKind.Client);
+            if (activity is not null)
```

Comment:
nit: in this kind of case I usually check for null and exit early, this way the rest of the method isn't indented